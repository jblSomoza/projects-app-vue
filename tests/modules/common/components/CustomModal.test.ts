import { mount } from "@vue/test-utils";
import CustomModal from '../../../../src/modules/common/components/CustomModal.vue';

describe('CustomModal', () => {
    test('renders dialog with default state', async () => {
        const wrapper = mount(CustomModal);
        const modal = wrapper.find('.modal');

        expect(modal.attributes('open')).toBeUndefined();
    });

    test('renders dialog with open state', async () => {
        const wrapper = mount(CustomModal, {
            props: {
                open: true
            }
        });
        const modal = wrapper.find('.modal');

        expect(modal.attributes('open')).toBeDefined();
    });

    test('renders dialog with header, body and footer slots', async () => {
        const wrapper = mount(CustomModal, {
            slots: {
                header: '<div>Header</div>',
                body: '<div>Body</div>',
                footer: '<div>Footer</div>'
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
        expect(wrapper.text()).toContain('Header');
        expect(wrapper.text()).toContain('Body');
        expect(wrapper.text()).toContain('Footer');

    });

    test('opens and closes dialog', async () => {
        const wrapper = mount(CustomModal);
        const modal = wrapper.find('.modal');

        expect(modal.attributes('open')).toBeUndefined();

        await wrapper.setProps({ open: true });

        expect(modal.attributes('open')).toBeDefined();

        await wrapper.setProps({ open: false });

        expect(modal.attributes('open')).toBeUndefined();
    });
});