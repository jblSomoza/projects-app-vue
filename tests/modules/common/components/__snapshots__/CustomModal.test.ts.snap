// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CustomModal > renders dialog with header, body and footer slots 1`] = `
"<dialog class="modal">
  <div class="modal-box">
    <div class="border-b border-blue-500">
      <!-- HEADER -->
      <div>Header</div>
    </div>
    <div class="my-5">
      <!-- BODY -->
      <div>Body</div>
    </div>
    <div class="border-t border-t-blue-500 pt-2">
      <!-- FOOTER -->
      <div>Footer</div>
    </div>
  </div>
</dialog>
<!--v-if-->"
`;
